### DISCUSSION TITLE
From the current codebase to our vision

### DISCUSSION THREAD

**Discussion by charlesBochet** [twentyhq/twenty#6805] [Engineering vision]
## Context

We've been working on Twenty for over a year, following a highly iterative process. We've explored various approaches, made mistakes, and in most cases, arrived at designs that work best.

Today, our engineering vision for the foundations is clear across most existing areas—about 95% of the code. However, the vision isn't always fully reflected in the codebase and is sometimes mixed with technical debt.

As we continue to develop product features, half of the team remains focused on refining the foundations. We believe this is crucial for Twenty's success, as reworking the core will become increasingly difficult as more use cases and features are added.

## Goal

This document aims to clarify the deltas between the vision and what we have in the current code base (ie list upcoming refactoring) and share it with the community.

## Frontend

- State Management / Component States: 
-- deprecate umbrella hooks useMyComponentStates: hooks should be as atomic as possible to avoid expensive computations during renders
-- complete and clean componentState API.

**Comment by BOHEUS** [comment]
Seems like it's quite outdated, if you disagree with me, please reopen discussion
